pkg/pulsar/subscription.go

```diff
@@ -71,73 +72,75 @@ type Subscriptions interface {
 }
 
 type subscriptions struct {
-	client   *client
+	client   *pulsarClient
+	request  *cli.Client
```

Comment:
why the type is `Client` but the variable is named `request`?

Reply:
The `Client` just send an HTTP request to a server and get the response. The name `request` shows what the client does.

Reply:
this is very confusing. If it is a request, then the type should be `Request`, not a `Client`. Also what is the difference between `pulsarClient` and `cli.Client`?

Reply:
`Client` is a base class that is used to send a request to a server.  And `pulsarClient` inherits the `Client`. 

Different client needs to use a different config.
`pulsarClient` created using pulsar-service-URL and API version and auth provider.
`bkClient` created using bk-service-URL and API version. 
I want to reuse the `Client` to send requests to the BK server. So I split the `Client` from `pulsarClient`.